tests/data/ar/CustomerWithConstructor.php

```diff
@@ -0,0 +1,51 @@
+<?php
+namespace yiiunit\data\ar;
+
+use yiiunit\framework\db\ActiveRecordTest;
+
+/**
+ * Class Customer
+ *
+ * @property int $id
+ * @property string $name
+ * @property string $email
+ * @property string $address
+ * @property int $status
+ *
+ * @property ProfileWithConstructor $profile
+ */
+class CustomerWithConstructor extends ActiveRecord
+{
+    const STATUS_ACTIVE = 1;
+    const STATUS_INACTIVE = 2;
+
+    public static function tableName()
+    {
+        return 'customer';
+    }
+
+    public function __construct($name, $email, $address)
+    {
+        $this->name = $name;
+        $this->email = $email;
+        $this->address = $address;
+        parent::__construct();
+    }
+
+    public static function instantiate($row = [])
+    {
+        return (new \ReflectionClass(static::className()))->newInstanceWithoutConstructor();
```

Comment:
> i think thats up for the developer to know how to handle the constructor and the instantiate.

If developer really wants to use own customized code and knows how to override `instantiate` method for working with his code that the developer can do it. If developer does not need it or does not know about this method existence he do not do it.

> what if the developer assumes some property, behavior or something got initialized by the constructor and start using it after a findOne() for example.

By default `findOne()` calls `instantiate($row)`. Method `instantiate` still calls `return new $class`. Logically that the developer can put this initialization code into `instantiate` method too if he has overwritten the method by his way.